src/backend/var_array.rs

```diff
@@ -0,0 +1,55 @@
+/// A C-style variable length array implemented as one allocation with
+/// a prepended header.
+use libpulse_sys::{pa_xrealloc};
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct VarArray<T>
```

Comment:
This was the trickiest and most unsafe part of the port. I'd love to do with out it.